tests/unit/lms/services/grouping_test.py

```diff
@@ -15,100 +14,80 @@ class TestGroupingService:
     CONTEXT_ID = "context_id"
     TOOL_CONSUMER_INSTANCE_GUID = "t_c_i_guid"
 
-    def test_upsert_inserts(self, svc, db_session):
-        # Start with no groupings
-        assert not db_session.query(Grouping).count()
+    def test_upsert_with_parents_inserts(self, svc, db_session):
+        course = factories.Course()
+        application_instance = factories.ApplicationInstance()
 
-        test_grouping = Grouping(
-            application_instance=factories.ApplicationInstance(),
-            authority_provided_id="ID",
+        # Start with no CanvasGroup
+        assert not db_session.query(CanvasGroup).count()
+
+        test_grouping = svc.upsert_with_parent(
+            tool_consumer_instance_guid=application_instance.tool_consumer_instance_guid,
```

Comment:
You can use the course's application instance (which I guess is what would happen in reality as well). Then you don't need the `application_instance` variable above:

```suggestion
            tool_consumer_instance_guid=course.application_instance.tool_consumer_instance_guid,
```